src/points/slice.ts

```diff
@@ -0,0 +1,58 @@
+import { PayloadAction, createSlice } from '@reduxjs/toolkit'
+import { ClaimHistory } from 'src/points/types'
+
+interface GetPointsHistorySucceededAction {
+  newPointsHistory: ClaimHistory[]
+  nextPageUrl: string | null
+}
+
+interface State {
+  pointsHistory: ClaimHistory[]
+  nextPageUrl: string | null
+  getHistoryStatus: 'idle' | 'loading-initial' | 'loading-more' | 'error'
+}
+
+const initialState: State = {
+  pointsHistory: [],
+  nextPageUrl: null,
+  getHistoryStatus: 'idle',
+}
+
+const slice = createSlice({
+  name: 'points',
+  initialState,
+  reducers: {
+    getInitialHistoryStarted: (state) => ({
```

Comment:
i realise the decision to have this explicit "initial" scenario may be related to the current designs. i started a discussion about that [in slack](https://valora-app.slack.com/archives/C0684TXDR3K/p1712309137807599). i guess i'm not entirely convinced that we should clear the history every time the points screen loads :/ that seems different to how we do the transaction feed...

taking it one step further, if the activity feed state needs to be reset every time the user launches the bottom sheet, we could not go through redux at all (why bother storing stuff in global state when it needs to be constantly cleared? we could instead use a hook that is naturally reset when unmounted)

i also wonder if we can simplify the state to only have 1 type of loading? 'loading-initial' is the same as 'loading' + pointsHistory is empty?

Reply:
I've simplified this a good deal. I've opted for an approach where, upon subsequent visits, we fetch the latest page of data (while displaying the old stale data) and as soon as that page has been fetched, we clear the stale data and overwrite it with the new, fresh history.

The alternative, where we _don't_ overwrite old data, but instead merge it with the newest page we've fetched, is more complicated and opens us up to potentially weird cases for users. For example, say a user views the points history once, and then the next time they view it is a month later, after earning lots of points. When they view the screen this second time, we'll fetch the latest page, and merge it with the old data, but there could be a big gap between the latest data and the old data. The history screen will render the new + old data side by side as if they're contiguous, and the in-between pages won't be fetched until the user drags down all the way _past_ the old data, which is weird. 

(One way to get around that is to _keep fetching_ fresh pages on subsequent viewings, until one of those pages contains a record that we already have in Redux, though I thought we might want to avoid that complexity.)